components/hab/src/command/pkg/uninstall/uninstall_impl.rs

```diff
@@ -330,12 +360,60 @@ fn maybe_delete<U>(ui: &mut U,
         }
         ExecutionStrategy::Run => {
             ui.status(Status::Deleting, &ident)?;
+            if uninstall_hook_mode == UninstallHookMode::Run {
+                maybe_run_uninstall_hook(ui, &install).await?;
+            }
             let pkg_dir = install.installed_path();
             do_clean_delete(&pkg_root_path, &pkg_dir)
         }
     }
 }
 
+async fn maybe_run_uninstall_hook<T>(ui: &mut T, package: &PackageInstall) -> Result<()>
+    where T: UIWriter
+{
+    let ident = package.ident();
+    let feature_flags = FeatureFlag::from_env(ui);
+    if let Some(ref hook) = UninstallHook::load(&ident.name,
+                                                &svc_hooks_path(ident.name.clone()),
+                                                &package.installed_path.join("hooks"),
+                                                feature_flags)
+    {
+        // We only want to run the uninstall hook if this is there are no other
+        // revisions of this package
```

Comment:
Seems like this should actually be the documentation comment for this function, perhaps with a bit more concrete detail about why it's important.